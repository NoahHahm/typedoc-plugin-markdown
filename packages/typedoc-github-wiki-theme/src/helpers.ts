export function parseUrl(url: string) {
  return encodeURI('../wiki/' + url.replace('.md', ''));
}
