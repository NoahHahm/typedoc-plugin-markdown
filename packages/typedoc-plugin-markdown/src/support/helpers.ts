import {
  ContainerReflection,
  DeclarationReflection,
  ProjectReflection,
  ReflectionKind,
  SignatureReflection,
} from 'typedoc';
import { backTicks } from './els';

/**
 * Determines if current signature is a constructor
 */
export const isConstructor = (
  signature: DeclarationReflection | SignatureReflection,
) => {
  return signature.parent?.kindOf(ReflectionKind.Constructor);
};

export function getIndexHeadingLevel(
  reflection: DeclarationReflection | ProjectReflection,
) {
  if (
    reflection.kindOf([
      ReflectionKind.Project,
      ReflectionKind.Module,
      ReflectionKind.Namespace,
    ]) ||
    reflection.hasOwnDocument
  ) {
    return 2;
  }
  return 4;
}

export function getGroupHeadingLevel(container: ContainerReflection) {
  return container.hasOwnDocument ? 2 : 4;
}

export function getReflectionHeadingLevel(
  reflection: DeclarationReflection | SignatureReflection,
) {
  if (reflection.hasOwnDocument) {
    return 1;
  }
  if (reflection.kindOf(ReflectionKind.Constructor)) {
    return reflection.parent?.hasOwnDocument ? 2 : 4;
  }
  return reflection.parent?.hasOwnDocument ? 3 : 5;
}

export function getMemberHeadingLevel(
  reflection: DeclarationReflection | SignatureReflection,
) {
  return getReflectionHeadingLevel(reflection) + 1;
}

export function getReflectionTitle(reflection: DeclarationReflection) {
  const md: string[] = [];

  md.push(reflection.name);

  if (reflection.typeParameters) {
    const typeParameters = reflection.typeParameters
      .map((typeParameter) => typeParameter.name)
      .join(', ');
    md.push(`\\<${typeParameters}\\>`);
  }

  return md.join('');
}

export function getFlags(reflection: DeclarationReflection) {
  if (reflection.flags?.length > 0 && !reflection.flags.isRest) {
    return reflection.flags.map((flag) => backTicks(flag)).join(' ');
  }
  return null;
}
