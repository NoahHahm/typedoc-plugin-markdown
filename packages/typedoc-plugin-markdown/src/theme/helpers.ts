/**
 *
 * A set of helpers that parses TypeDoc models to be consumed by theme resources.
 *
 * @module
 */

import {
  DeclarationReflection,
  ProjectReflection,
  ReflectionKind,
} from 'typedoc';
import { backTicks } from '../support/elements';
import { escapeChars } from '../support/utils';

export function getDeclarationType(declaration: DeclarationReflection) {
  if (declaration.getSignature) {
    return declaration.getSignature.type;
  }
  if (declaration.setSignature) {
    return declaration.setSignature.type;
  }
  return declaration.type;
}

export function getProjectDisplayName(
  project: ProjectReflection,
  includeVersion: boolean,
): string {
  const version =
    includeVersion && project.packageVersion
      ? ` - v${project.packageVersion}`
      : '';
  return `${project.name}${version ? version : ''}`;
}

export function hasIndex(
  reflection: DeclarationReflection | ProjectReflection,
) {
  return reflection.groups?.some((group) => group.allChildrenHaveOwnDocument());
}

export function hasTOC(reflection: DeclarationReflection | ProjectReflection) {
  return (
    reflection.kindOf([
      ReflectionKind.Project,
      ReflectionKind.Module,
      ReflectionKind.Namespace,
    ]) &&
    reflection.groups?.some((group) => !group.allChildrenHaveOwnDocument())
  );
}

export function isGroupKind(reflection: DeclarationReflection) {
  return reflection.kindOf([
    ReflectionKind.Class,
    ReflectionKind.Interface,
    ReflectionKind.Enum,
    ReflectionKind.Function,
    ReflectionKind.Variable,
    ReflectionKind.TypeAlias,
  ]);
}

export function getModifier(reflection: DeclarationReflection) {
  if (reflection.flags.isAbstract) {
    return 'abstract';
  }
  if (reflection.flags.isPrivate) {
    return 'private';
  }
  if (reflection.flags.isReadonly) {
    return 'readonly';
  }
  if (reflection.flags.isStatic) {
    return 'static';
  }
  if (reflection.getSignature) {
    return 'get';
  }
  if (reflection.setSignature) {
    return 'set';
  }
  return null;
}

export const KEYWORD_MAP = {
  [ReflectionKind.Class]: 'class',
  [ReflectionKind.Interface]: 'interface',
  [ReflectionKind.Enum]: 'enum',
  [ReflectionKind.TypeAlias]: 'type',
};

export function getMemberTitle(reflection: DeclarationReflection) {
  const md: string[] = [];

  const name: string[] = [];

  const flagsToInclude = ['Abstract'];

  const flags = reflection.flags.filter((flag) =>
    flagsToInclude.includes(flag),
  );

  if (flags.length) {
    name.push(flags.map((flag) => backTicks(flag.toLowerCase())).join(' '));
    name.push(' ');
  }

  name.push(`${escapeChars(reflection.name)}`);

  if (reflection.typeParameters) {
    const typeParameters = reflection.typeParameters
      .map((typeParameter) => typeParameter.name)
      .join(', ');
    name.push(`${backTicks(`<${typeParameters}>`)}`);
  }

  md.push(name.join(''));

  return md.join(': ');
}
