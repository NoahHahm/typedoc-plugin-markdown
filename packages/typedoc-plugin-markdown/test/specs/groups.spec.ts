import {
  FixtureOutputDir,
  FixtureOutputFileStrategy,
} from '../__utils__/config';
import { expectFileToEqual } from '../__utils__/helpers';

describe(`Groups`, () => {
  test(`should compile index page for project`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      [FixtureOutputFileStrategy.Modules, FixtureOutputFileStrategy.Members],
      ['API.md', 'index.md'],
    );
  });

  test(`should compile index page for monorepo`, () => {
    expectFileToEqual(
      FixtureOutputDir.Packages,
      FixtureOutputFileStrategy.Members,
      ['API.md'],
    );
  });

  test(`should compile basic module index page`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      FixtureOutputFileStrategy.Modules,
      'basic.md',
    );

    expectFileToEqual(
      FixtureOutputDir.Groups,
      FixtureOutputFileStrategy.Members,
      ['basic/README.md', 'basic/index.md'],
    );
  });

  test(`should compile module index with namespaces index page`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      [FixtureOutputFileStrategy.Modules, FixtureOutputFileStrategy.Members],
      ['has-namespaces/README.md', 'has-namespaces/index.md'],
    );
  });

  test(`should compile module index for a namespace`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      [FixtureOutputFileStrategy.Modules, FixtureOutputFileStrategy.Members],
      [
        'has-namespaces/namespaces/Namespace_A_/README.md',
        'has-namespaces/namespaces/Namespace_A_/index.md',
      ],
    );
  });

  test(`should compile categories group page`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      FixtureOutputFileStrategy.Modules,
      'has-categories.md',
    );
  });

  test(`should compile categories index page`, () => {
    expectFileToEqual(
      FixtureOutputDir.Groups,
      FixtureOutputFileStrategy.Members,
      ['has-categories/README.md', 'has-categories/index.md'],
      2,
    );
  });
});
