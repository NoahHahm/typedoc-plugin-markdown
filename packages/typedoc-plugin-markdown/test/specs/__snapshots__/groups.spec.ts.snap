// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups should compile basic module index page: (Output File Strategy "members") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](../README.md) ∙ [API](../modules.md)

***

[API](../modules.md) > basic

# Module: basic

A basic module

## Index

### Enumerations

- [EnumA](enumerations/EnumA.md)
- [EnumB](enumerations/EnumB.md)

### Classes

- [ClassA](classes/ClassA.md)
- [ClassB](classes/ClassB.md)

### Interfaces

- [IterfaceA](interfaces/IterfaceA.md)
- [IterfaceB](interfaces/IterfaceB.md)

### Type Aliases

- [TypeA](type-aliases/TypeA.md)
- [TypeB](type-aliases/TypeB.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile basic module index page: (Output File Strategy "members") (Option Group "2") 1`] = `
"# Module: basic

A basic module

## Enumerations

| Enumeration | Description |
| :------ | :------ |
| [EnumA](enumerations/EnumA.md) | - |
| [EnumB](enumerations/EnumB.md) | - |

## Classes

| Class | Description |
| :------ | :------ |
| [ClassA](classes/ClassA.md) | - |
| [ClassB](classes/ClassB.md) | - |

## Interfaces

| Interface | Description |
| :------ | :------ |
| [IterfaceA](interfaces/IterfaceA.md) | - |
| [IterfaceB](interfaces/IterfaceB.md) | - |

<a id="typea" name="typea"></a>

## TypeA

> **TypeA**: \`string\`

***

<a id="typeb" name="typeb"></a>

## TypeB

> **TypeB**: \`string\`

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile basic module index page: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](README.md) ∙ [API](modules.md)

***

[API](modules.md) > basic

# Module: basic

A basic module

## Contents

- [Enumerations](basic.md#enumerations)
  - [EnumA](basic.md#enuma)
  - [EnumB](basic.md#enumb)
- [Classes](basic.md#classes)
  - [ClassA](basic.md#classa)
  - [ClassB](basic.md#classb)
- [Interfaces](basic.md#interfaces)
  - [IterfaceA](basic.md#iterfacea)
  - [IterfaceB](basic.md#iterfaceb)
- [Type Aliases](basic.md#type-aliases)
  - [TypeA](basic.md#typea)
  - [TypeB](basic.md#typeb)

## Enumerations

### EnumA

***

### EnumB

## Classes

### ClassA

#### Constructors

##### new ClassA()

> **new ClassA**(): [\`ClassA\`](basic.md#classa)

***

### ClassB

#### Constructors

##### new ClassB()

> **new ClassB**(): [\`ClassB\`](basic.md#classb)

## Interfaces

### IterfaceA

#### Properties

##### propA

> **propA**: \`string\`

##### propB

> **propB**: \`string\`

***

### IterfaceB

## Type Aliases

### TypeA

> **TypeA**: \`string\`

***

### TypeB

> **TypeB**: \`string\`

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile basic module index page: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# Module: basic

A basic module

<a id="enuma" name="enuma"></a>

## EnumA

***

<a id="enumb" name="enumb"></a>

## EnumB

***

<a id="classa" name="classa"></a>

## ClassA

### Constructors

<a id="constructors" name="constructors"></a>

#### new ClassA()

> **new ClassA**(): [\`ClassA\`](basic.md#classa)

***

<a id="classb" name="classb"></a>

## ClassB

### Constructors

<a id="constructors-1" name="constructors-1"></a>

#### new ClassB()

> **new ClassB**(): [\`ClassB\`](basic.md#classb)

***

<a id="iterfacea" name="iterfacea"></a>

## IterfaceA

### Properties

<a id="propa" name="propa"></a>

#### propA

> **propA**: \`string\`

<a id="propb" name="propb"></a>

#### propB

> **propB**: \`string\`

***

<a id="iterfaceb" name="iterfaceb"></a>

## IterfaceB

***

<a id="typea" name="typea"></a>

## TypeA

> **TypeA**: \`string\`

***

<a id="typeb" name="typeb"></a>

## TypeB

> **TypeB**: \`string\`

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile categories group page: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](README.md) ∙ [API](modules.md)

***

[API](modules.md) > has-categories

# Module: has-categories

A module that contains categories

## Contents

- [Enumerations](has-categories.md#enumerations)
  - [Category A](has-categories.md#category-a)
  - [Category B](has-categories.md#category-b)
  - [Other](has-categories.md#other)
- [Interfaces](has-categories.md#interfaces)
  - [Category A](has-categories.md#category-a-1)
  - [Category B](has-categories.md#category-b-1)
  - [Other](has-categories.md#other-1)

## Enumerations

### Category A

#### CategoryAEnum1

***

#### CategoryAEnum2

### Category B

#### CategoryBEnum1

***

#### CategoryBEnum2

### Other

#### UnCategorizedEnum

## Interfaces

### Category A

#### CategoryAInterface1

***

#### CategoryAInterface2

### Category B

#### CategoryBInterface1

***

#### CategoryBInterface2

### Other

#### UnCategorizedInterace

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile categories group page: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# Module: has-categories

A module that contains categories

## Category A

<a id="categoryaenum1" name="categoryaenum1"></a>

### CategoryAEnum1

***

<a id="categoryaenum2" name="categoryaenum2"></a>

### CategoryAEnum2

***

<a id="categoryainterface1" name="categoryainterface1"></a>

### CategoryAInterface1

***

<a id="categoryainterface2" name="categoryainterface2"></a>

### CategoryAInterface2

## Category B

<a id="categorybenum1" name="categorybenum1"></a>

### CategoryBEnum1

***

<a id="categorybenum2" name="categorybenum2"></a>

### CategoryBEnum2

***

<a id="categorybinterface1" name="categorybinterface1"></a>

### CategoryBInterface1

***

<a id="categorybinterface2" name="categorybinterface2"></a>

### CategoryBInterface2

## Other

<a id="uncategorizedenum" name="uncategorizedenum"></a>

### UnCategorizedEnum

***

<a id="uncategorizedinterace" name="uncategorizedinterace"></a>

### UnCategorizedInterace

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile categories index page: (Output File Strategy "members") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](../README.md) ∙ [API](../modules.md)

***

[API](../modules.md) > has-categories

# Module: has-categories

A module that contains categories

## Index

### Enumerations

#### Category A

- [CategoryAEnum1](enumerations/CategoryAEnum1.md)
- [CategoryAEnum2](enumerations/CategoryAEnum2.md)

#### Category B

- [CategoryBEnum1](enumerations/CategoryBEnum1.md)
- [CategoryBEnum2](enumerations/CategoryBEnum2.md)

#### Other

- [UnCategorizedEnum](enumerations/UnCategorizedEnum.md)

### Interfaces

#### Category A

- [CategoryAInterface1](interfaces/CategoryAInterface1.md)
- [CategoryAInterface2](interfaces/CategoryAInterface2.md)

#### Category B

- [CategoryBInterface1](interfaces/CategoryBInterface1.md)
- [CategoryBInterface2](interfaces/CategoryBInterface2.md)

#### Other

- [UnCategorizedInterace](interfaces/UnCategorizedInterace.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile categories index page: (Output File Strategy "members") (Option Group "2") 1`] = `
"# Module: has-categories

A module that contains categories

## Index

### Category A

| Enumeration | Description |
| :------ | :------ |
| [CategoryAEnum1](enumerations/CategoryAEnum1.md) | - |
| [CategoryAEnum2](enumerations/CategoryAEnum2.md) | - |
| [CategoryAInterface1](interfaces/CategoryAInterface1.md) | - |
| [CategoryAInterface2](interfaces/CategoryAInterface2.md) | - |

### Category B

| Enumeration | Description |
| :------ | :------ |
| [CategoryBEnum1](enumerations/CategoryBEnum1.md) | - |
| [CategoryBEnum2](enumerations/CategoryBEnum2.md) | - |
| [CategoryBInterface1](interfaces/CategoryBInterface1.md) | - |
| [CategoryBInterface2](interfaces/CategoryBInterface2.md) | - |

### Other

| Enumeration | Description |
| :------ | :------ |
| [UnCategorizedEnum](enumerations/UnCategorizedEnum.md) | - |
| [UnCategorizedInterace](interfaces/UnCategorizedInterace.md) | - |

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile index page for monorepo: (Output File Strategy "members") (Option Group "1") 1`] = `
"**Documentation** ∙ [README](README.md) ∙ [API](modules.md)

***

# API

## Index

- [package-1](package-1/README.md)
- [package-2](package-2/README.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile index page for project: (Output File Strategy "members") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](README.md) ∙ [API](modules.md)

***

# API

## Index

### Modules

- [basic](basic/README.md)
- [has-categories](has-categories/README.md)
- [has-namespaces](has-namespaces/README.md)
- [index](index/README.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile index page for project: (Output File Strategy "members") (Option Group "2") 1`] = `
"# API

## Index

### Modules

| Module | Description |
| :------ | :------ |
| [basic](basic/index.md) | A basic module |
| [has-categories](has-categories/index.md) | A module that contains categories |
| [has-namespaces](has-namespaces/index.md) | A module that contains namespaces |
| [index](module.index/index.md) | A module named index |

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile index page for project: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](README.md) ∙ [API](modules.md)

***

# API

## Index

### Modules

- [basic](basic.md)
- [has-categories](has-categories.md)
- [has-namespaces](has-namespaces/README.md)
- [index](index.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile index page for project: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# API

## Index

### Modules

| Module | Description |
| :------ | :------ |
| [basic](basic.md) | A basic module |
| [has-categories](has-categories.md) | A module that contains categories |
| [has-namespaces](has-namespaces/index.md) | A module that contains namespaces |
| [index](module.index.md) | A module named index |

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile module index for a namespace: (Output File Strategy "members") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](../../../README.md) ∙ [API](../../../modules.md)

***

[API](../../../modules.md) > [has-namespaces](../../README.md) > \\_Namespace\\_A\\_

# Namespace: \\_Namespace\\_A\\_

## Index

### Namespaces

- [NestedNamespace](namespaces/NestedNamespace/README.md)

### Interfaces

- [NamespaceInterface](interfaces/NamespaceInterface.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile module index for a namespace: (Output File Strategy "members") (Option Group "2") 1`] = `
"# Namespace: \\_Namespace\\_A\\_

## Index

### Namespaces

| Namespace | Description |
| :------ | :------ |
| [NestedNamespace](namespaces/NestedNamespace/index.md) | - |

### Interfaces

| Interface | Description |
| :------ | :------ |
| [NamespaceInterface](interfaces/NamespaceInterface.md) | - |

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile module index for a namespace: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](../../../README.md) ∙ [API](../../../modules.md)

***

[API](../../../modules.md) > [has-namespaces](../../README.md) > \\_Namespace\\_A\\_

# Namespace: \\_Namespace\\_A\\_

## Contents

- [Namespaces](README.md#namespaces)
- [Interfaces](README.md#interfaces)
  - [NamespaceInterface](README.md#namespaceinterface)

## Namespaces

- [NestedNamespace](namespaces/NestedNamespace.md)

## Interfaces

### NamespaceInterface

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile module index for a namespace: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# Namespace: \\_Namespace\\_A\\_

## Namespaces

| Namespace | Description |
| :------ | :------ |
| [NestedNamespace](namespaces/NestedNamespace.md) | - |

<a id="namespaceinterface" name="namespaceinterface"></a>

## NamespaceInterface

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile module index with namespaces index page: (Output File Strategy "members") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](../README.md) ∙ [API](../modules.md)

***

[API](../modules.md) > has-namespaces

# Module: has-namespaces

A module that contains namespaces

## Index

### Namespaces

- [NamespaceB](namespaces/NamespaceB/README.md)
- [\\_Namespace\\_A\\_](namespaces/Namespace_A_/README.md)

### Enumerations

- [EnumB](enumerations/EnumB.md)
- [\\_Enum\\_A\\_](enumerations/Enum_A_.md)

### Classes

- [ClassB](classes/ClassB.md)
- [\\_\\_Class\\_A\\_](classes/Class_A_.md)

### Interfaces

- [InterfaceB](interfaces/InterfaceB.md)
- [\\_Interface\\_A\\_](interfaces/Interface_A_.md)

### Type Aliases

- [TypeB](type-aliases/TypeB.md)
- [\\_Type\\_A\\_](type-aliases/Type_A_.md)

### Variables

- [\\_variable\\_A\\_](variables/variable_A_.md)
- [variableB](variables/variableB.md)

### Functions

- [\\_function\\_A\\_](functions/function_A_.md)
- [functionB](functions/functionB.md)

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile module index with namespaces index page: (Output File Strategy "members") (Option Group "2") 1`] = `
"# Module: has-namespaces

A module that contains namespaces

## Namespaces

| Namespace | Description |
| :------ | :------ |
| [NamespaceB](namespaces/NamespaceB/index.md) | - |
| [\\_Namespace\\_A\\_](namespaces/Namespace_A_/index.md) | - |

## Enumerations

| Enumeration | Description |
| :------ | :------ |
| [EnumB](enumerations/EnumB.md) | - |
| [\\_Enum\\_A\\_](enumerations/Enum_A_.md) | - |

## Classes

| Class | Description |
| :------ | :------ |
| [ClassB](classes/ClassB.md) | - |
| [\\_\\_Class\\_A\\_](classes/Class_A_.md) | - |

## Interfaces

| Interface | Description |
| :------ | :------ |
| [InterfaceB](interfaces/InterfaceB.md) | - |
| [\\_Interface\\_A\\_](interfaces/Interface_A_.md) | - |

<a id="typeb" name="typeb"></a>

## TypeB

> **TypeB**: \`string\`

***

<a id="type-a" name="type-a"></a>

## \\_Type\\_A\\_

> **\\_Type\\_A\\_**: \`string\`

***

<a id="variable-a" name="variable-a"></a>

## \\_variable\\_A\\_

> **\`const\`** **\\_variable\\_A\\_**: \`true\` = \`true\`

***

<a id="variableb" name="variableb"></a>

## variableB

> **\`const\`** **variableB**: \`true\` = \`true\`

***

<a id="function-a" name="function-a"></a>

## \\_function\\_A\\_()

> **\\_function\\_A\\_**(): \`void\`

***

<a id="functionb" name="functionb"></a>

## functionB()

> **functionB**(): \`void\`

CUSTOM FOOTER HERE
"
`;

exports[`Groups should compile module index with namespaces index page: (Output File Strategy "modules") (Option Group "1") 1`] = `
"[**typedoc-stubs - v0.0**](http://www.google.com) ∙ [README](../README.md) ∙ [API](../modules.md)

***

[API](../modules.md) > has-namespaces

# Module: has-namespaces

A module that contains namespaces

## Contents

- [Namespaces](README.md#namespaces)
- [Enumerations](README.md#enumerations)
  - [EnumB](README.md#enumb)
  - [\\_Enum\\_A\\_](README.md#enum-a)
- [Classes](README.md#classes)
  - [ClassB](README.md#classb)
  - [\\_\\_Class\\_A\\_](README.md#class-a)
- [Interfaces](README.md#interfaces)
  - [InterfaceB](README.md#interfaceb)
  - [\\_Interface\\_A\\_](README.md#interface-a)
- [Type Aliases](README.md#type-aliases)
  - [TypeB](README.md#typeb)
  - [\\_Type\\_A\\_](README.md#type-a)
- [Variables](README.md#variables)
  - [\\_variable\\_A\\_](README.md#variable-a)
  - [variableB](README.md#variableb)
- [Functions](README.md#functions)
  - [\\_function\\_A\\_()](README.md#function-a)
  - [functionB()](README.md#functionb)

## Namespaces

- [NamespaceB](namespaces/NamespaceB.md)
- [\\_Namespace\\_A\\_](namespaces/Namespace_A_/README.md)

## Enumerations

### EnumB

***

### \\_Enum\\_A\\_

## Classes

### ClassB

#### Constructors

##### new ClassB()

> **new ClassB**(): [\`ClassB\`](README.md#classb)

***

### \\_\\_Class\\_A\\_

#### Constructors

##### new \\_\\_Class\\_A\\_()

> **new \\_\\_Class\\_A\\_**(): [\`__Class_A_\`](README.md#class-a)

## Interfaces

### InterfaceB

***

### \\_Interface\\_A\\_

## Type Aliases

### TypeB

> **TypeB**: \`string\`

***

### \\_Type\\_A\\_

> **\\_Type\\_A\\_**: \`string\`

## Variables

### \\_variable\\_A\\_

> **\`const\`** **\\_variable\\_A\\_**: \`true\` = \`true\`

***

### variableB

> **\`const\`** **variableB**: \`true\` = \`true\`

## Functions

### \\_function\\_A\\_()

> **\\_function\\_A\\_**(): \`void\`

***

### functionB()

> **functionB**(): \`void\`

***

Generated using [typedoc-plugin-markdown](https://www.npmjs.com/package/typedoc-plugin-markdown) and [TypeDoc](https://typedoc.org/)
"
`;

exports[`Groups should compile module index with namespaces index page: (Output File Strategy "modules") (Option Group "2") 1`] = `
"# Module: has-namespaces

A module that contains namespaces

## Namespaces

| Namespace | Description |
| :------ | :------ |
| [NamespaceB](namespaces/NamespaceB.md) | - |
| [\\_Namespace\\_A\\_](namespaces/Namespace_A_/index.md) | - |

<a id="enumb" name="enumb"></a>

## EnumB

***

<a id="enum-a" name="enum-a"></a>

## \\_Enum\\_A\\_

***

<a id="classb" name="classb"></a>

## ClassB

### Constructors

<a id="constructors" name="constructors"></a>

#### new ClassB()

> **new ClassB**(): [\`ClassB\`](index.md#classb)

***

<a id="class-a" name="class-a"></a>

## \\_\\_Class\\_A\\_

### Constructors

<a id="constructors-1" name="constructors-1"></a>

#### new \\_\\_Class\\_A\\_()

> **new \\_\\_Class\\_A\\_**(): [\`__Class_A_\`](index.md#class-a)

***

<a id="interfaceb" name="interfaceb"></a>

## InterfaceB

***

<a id="interface-a" name="interface-a"></a>

## \\_Interface\\_A\\_

***

<a id="typeb" name="typeb"></a>

## TypeB

> **TypeB**: \`string\`

***

<a id="type-a" name="type-a"></a>

## \\_Type\\_A\\_

> **\\_Type\\_A\\_**: \`string\`

***

<a id="variable-a" name="variable-a"></a>

## \\_variable\\_A\\_

> **\`const\`** **\\_variable\\_A\\_**: \`true\` = \`true\`

***

<a id="variableb" name="variableb"></a>

## variableB

> **\`const\`** **variableB**: \`true\` = \`true\`

***

<a id="function-a" name="function-a"></a>

## \\_function\\_A\\_()

> **\\_function\\_A\\_**(): \`void\`

***

<a id="functionb" name="functionb"></a>

## functionB()

> **functionB**(): \`void\`

CUSTOM FOOTER HERE
"
`;
